2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -113,68 +113,51 @@ function mongo_initiate() {
   mongo admin --eval "${config};rs.initiate(config);${mongo_wait}"
 }
 
-# get the address of the current primary member
-function mongo_primary_member_addr() {
-  local rc=0
-
-  endpoints | grep -v "$(container_addr)" |
-  (
-    while read mongo_node; do
-      cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
-
-      # Trying to find IP:PORT in output and filter out error message because mongo prints it to stdout
-      ip_and_port_regexp='[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+:[0-9]\+'
-      if addr="$(echo "$cmd_output" | grep -x "$ip_and_port_regexp")"; then
-        echo -n "$addr"
-        exit 0
-      fi
-
-      echo >&2 "Cannot get address of primary from $mongo_node node: $cmd_output"
-    done
-
-    exit 1
-  ) || rc=$?
-
-  if [ $rc -ne 0 ]; then
-    echo >&2 "Cannot get address of primary node: after checking all nodes we don't have the address"
+# replset_addr return the address of the current replSet
+function replset_addr() {
+  local current_endpoints
+  current_endpoints="$(endpoints)"
+  if [ -z "${current_endpoints}" ]; then
+    echo >&2 "Cannot get address of replica set: no nodes are listed in service"
     return 1
   fi
+  echo "${MONGODB_REPLICA_NAME}/$(echo ${current_endpoints//[[:space:]]/,}"
 }
 
 # mongo_remove removes the current MongoDB from the cluster
 function mongo_remove() {
-  local primary_addr
+  local host
   # if we cannot determine the IP address of the primary, exit without an error
   # to allow callers to proceed with their logic
-  primary_addr="$(mongo_primary_member_addr || true)"
-  if [ -z "$primary_addr" ]; then
+  host="$(replset_addr || true)"
+  if [ -z "$host" ]; then
     return
   fi
 
   local mongo_addr
   mongo_addr="$(mongo_addr)"
 
-  echo "=> Removing ${mongo_addr} on ${primary_addr} ..."
+  echo "=> Removing ${mongo_addr} on ${host} ..."
```

Comment:
This message now will look a bit weird on the console output, as it will read something like:

```
=> Removing 81.114.31.36 on rs0/81.114.31.36,81.114.31.35,81.114.31.37
```

Maybe we should omit the host part and just say something like:

```bash
  echo "=> Removing ${mongo_addr} from replica set ..."
```

WDYT?